File: python/Creational/Abstract Factory/MazeFactory.py
from abc import ABC, abstractmethod

class MazeFactory(ABC):
    @abstractmethod
    def make_maze(self):
        pass

    @abstractmethod
    def make_wall(self):
        pass

    @abstractmethod
    def make_room(self, room_no):
        pass

    @abstractmethod
    def make_door(self, room1, room2):
        pass

File: js/Creationals/Abstract Factory/MazeFactory.js
export class MazeFactory {
    makeMaze() {
        throw new Error("This method is abstract and must be implemented");
    }

    makeWall() {
        throw new Error("This method is abstract and must be implemented");
    }

    makeRoom(roomNo) {
        throw new Error("This method is abstract and must be implemented");
    }

    makeDoor(room1, room2) {
        throw new Error("This method is abstract and must be implemented");
    }
}
